mitodl/ocw-studio
# [proposal] try using swc for building assets

#### Pre-Flight checklist


#### What are the relevant tickets?

none

#### What's this PR do?

This is a dev tooling change which I want to propose and see what everyone things. It basically swaps out babel and the typescript compiler in our build / test toolchain for [swc](https://swc.rs/), which is a series of tools built in Rust for compiling and bundling web things. Because the tools are written in Rust and compiled to native code they are generally much faster than tools written in JS (like babel, the typescript compiler, etc) so our builds and tests run quite a bit quickly.

In my testing I observed the tests running in a little under half the time, and CI builds seem to go from ~4 minutes to ~2.5. Nice!

#### How should this be manually tested?

Test a few things:

- the build should be passing!
- you should be able to run tests locally
- the local dev build (in the `watch` container) should work without issue

My thought is that it would be good to get some feedback on this. I like the speed increase, but I'm not 100% sure we want to adopt it. What do y'all think?
